executor/show.go

```diff
@@ -583,153 +583,166 @@ func (e *ShowExec) fetchShowCreateTable() error {
-			if c.Length != types.UnspecifiedLength {
-				colInfo = fmt.Sprintf("%s(%s)", colInfo, strconv.Itoa(c.Length))
-			}
-			cols = append(cols, colInfo)
-		}
-		fmt.Fprintf(&buf, "(%s)", strings.Join(cols, ","))
-		if i != len(publicIndices)-1 {
-			buf.WriteString(",\n")
+		// Displayed if the compression typed is set.
+		if len(tb.Meta().Compression) != 0 {
+			fmt.Fprintf(&buf, " COMPRESSION='%s'", tb.Meta().Compression)
 		}
-	}
 
-	buf.WriteString("\n")
+		// add partition info here.
+		appendPartitionInfo(tb.Meta().Partition, &buf)
 
-	buf.WriteString(") ENGINE=InnoDB")
-	charsetName := tb.Meta().Charset
-	if len(charsetName) == 0 {
-		charsetName = mysql.DefaultCharset
-	}
-	collate := tb.Meta().Collate
-	// Set default collate if collate is not specified.
-	if len(collate) == 0 {
-		collate = getDefaultCollate(charsetName)
-	}
-	// Because we only support case sensitive utf8_bin collate, we need to explicitly set the default charset and collation
-	// to make it work on MySQL server which has default collate utf8_general_ci.
-	if len(collate) == 0 {
-		// If we can not find default collate for the given charset,
-		// do not show the collate part.
-		fmt.Fprintf(&buf, " DEFAULT CHARSET=%s", charsetName)
-	} else {
-		fmt.Fprintf(&buf, " DEFAULT CHARSET=%s COLLATE=%s", charsetName, collate)
-	}
-
-	// Displayed if the compression typed is set.
-	if len(tb.Meta().Compression) != 0 {
-		fmt.Fprintf(&buf, " COMPRESSION='%s'", tb.Meta().Compression)
-	}
+		if hasAutoIncID {
+			autoIncID, err := tb.Allocator(e.ctx).NextGlobalAutoID(tb.Meta().ID)
+			if err != nil {
+				return errors.Trace(err)
+			}
+			// It's campatible with MySQL.
+			if autoIncID > 1 {
+				fmt.Fprintf(&buf, " AUTO_INCREMENT=%d", autoIncID)
+			}
+		}
 
-	// add partition info here.
-	appendPartitionInfo(tb.Meta().Partition, &buf)
+		if tb.Meta().ShardRowIDBits > 0 {
+			fmt.Fprintf(&buf, "/*!90000 SHARD_ROW_ID_BITS=%d */", tb.Meta().ShardRowIDBits)
+		}
 
-	if hasAutoIncID {
-		autoIncID, err := tb.Allocator(e.ctx).NextGlobalAutoID(tb.Meta().ID)
-		if err != nil {
-			return errors.Trace(err)
+		if len(tb.Meta().Comment) > 0 {
+			fmt.Fprintf(&buf, " COMMENT='%s'", format.OutputFormat(tb.Meta().Comment))
 		}
-		// It's campatible with MySQL.
-		if autoIncID > 1 {
-			fmt.Fprintf(&buf, " AUTO_INCREMENT=%d", autoIncID)
+	} else {
+		fmt.Fprintf(&buf, "CREATE ALGORITHM=%s ", tb.Meta().View.Algorithm.String())
+		fmt.Fprintf(&buf, "DEFINER=%s@%s ", escape(model.NewCIStr(tb.Meta().View.Definer.Username), sqlMode), escape(model.NewCIStr(tb.Meta().View.Definer.Hostname), sqlMode))
+		fmt.Fprintf(&buf, "SQL SECURITY %s ", tb.Meta().View.Security.String())
+		fmt.Fprintf(&buf, "VIEW %s (", escape(tb.Meta().Name, sqlMode))
+		for i, col := range tb.Meta().Columns {
+			fmt.Fprintf(&buf, "%s", escape(col.Name, sqlMode))
+			if i < len(tb.Meta().Columns)-1 {
+				fmt.Fprintf(&buf, ", ")
+			}
 		}
+		fmt.Fprintf(&buf, ") AS %s", tb.Meta().View.SelectStmt)
```

Comment:
This is the output of MySQL:
```
mysql> show create table v_test;
+--------+-------------------------------------------------------------------------------------------------------------------------------------------+----------------------+----------------------+
| View   | Create View                                                                                                                               | character_set_client | collation_connection |
+--------+-------------------------------------------------------------------------------------------------------------------------------------------+----------------------+----------------------+
| v_test | CREATE ALGORITHM=UNDEFINED DEFINER=`root`@`localhost` SQL SECURITY DEFINER VIEW `v_test` AS select `t`.`a` AS `a`,`t`.`b` AS `b` from `t` | utf8                 | utf8_general_ci      |
+--------+-------------------------------------------------------------------------------------------------------------------------------------------+----------------------+----------------------+
1 row in set (0.00 sec)

mysql> show create table t;
+-------+------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                     |
+-------+------------------------------------------------------------------------------------------------------------------+
| t     | CREATE TABLE `t` (
  `a` int(11) DEFAULT NULL,
  `b` int(11) DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=latin1 |
+-------+------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```
seems the number of columns and column names are different between normal table and view.

Reply:
@AndrewDi Please take a look at this comment.